lib/pbench/server/api/resources/upload_api.py

```diff
@@ -348,6 +337,22 @@ def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Respon
                         f"Failed to write .md5 file '{md5_full_path}'",
                     )
 
+            try:
+                metadata = Tarball.get_metadata(tar_full_path)
+                controller = metadata["run"]["controller"]
```

Comment:
Should this be in a try-except block of its own? That way, if a key is missing, we can propagate an appropriate response back to the user instead of the generic response on line 350.

Reply:
Let's resist the urge to have too many `try` blocks!  🥴  If we need to catch it specially, we should be able to do so by adding a `KeyError` clause to the existing `try` block.  However, @npalaska, why do you think that the existing handling at line 350 is insufficient?  I _think_ it will explicitly report that the `"controller"` key is missing if it is omitted from the metadata.

Reply:
Yeah, I guess that's sufficient.